src/codegen/compile.ml

```diff
@@ -7253,12 +7286,12 @@ module Serialization = struct
                We update the memo location here so that loops work
             *)
             get_thing ^^ set_result ^^
-            get_memo ^^ get_result ^^ store_word32 ^^
-            get_memo ^^ compile_add_const 4L ^^ Blob.lit env (typ_hash t) ^^ store_word32
+            get_memo ^^ get_result ^^ write_compressed_pointer env ^^
+            get_memo ^^ compile_add_const 4L ^^ Blob.lit env (typ_hash t) ^^ write_compressed_pointer env
           )
           end begin
           (* Decoded before. Check type hash *)
-          ReadBuf.read_word32 env get_data_buf ^^ Blob.lit env (typ_hash t) ^^
+          read_compressed_pointer env get_data_buf ^^ Blob.lit env (typ_hash t) ^^
           Blob.compare env (Some Operator.EqOp) ^^
```

Comment:
On master, this Blob.compare is actually just a pointer equality test, which could be cheaper. Can we not get away with that anymore? Is it because we don't so static allocation, and these two Blob.lit occurrences might reside at different addresses, despite the same hash?

Reply:
Unfortunately, yes, the two blobs are equal but no longer the same object because static allocation is no longer used. If we would deduplicate multiple occurrences of the same blob content in future, the corresponding singleton can be compared by equality. Do you think the Blob comparison is too expensive in terms of instruction costs on upgrade?

Reply:
I think this is just a sanity check, we maybe we could disable it in production. Or, maybe we should just statically compute a small word-size hash from (type_hash t) and use that instead.

Reply:
Actually, maybe we can just use Lib.Crc.crc32 (in src/lib) to pre-compute a small hash of (typ_hash t) and check against that.